### PR TITLE
updating teamsnap-ui getting started documentation

### PR DESCRIPTION
**Pull request by williamdingwall**
## In Progress....

Updating the ui-patterns documentation to include the some of the basic design elements that build up to the components, make it a little easier for a new dev diving in, stating some of the goals of TS-UI, re-organizing the navigation, etc..

A few screenshots:
Screen Shot 2019-05-24 at 7 28 32 AM
Screen Shot 2019-05-24 at 7 28 44 AM

Navigation Updates: From top to bottom I wanted to lead people into TeamSnap-UI a little differently by starting to explain why we're doing it and how to get started using right up front, as well as bring some of the more important aspects about to the top too (Demos/Sandbox). Then follow that up with design elements to explain some underlying structure of the site design. Next utilities since that is probably the most important thing to wrap your head around as a dev picking this up. Then components, and Help Docs focused on contributing.  So the general flow from top to bottom is get someone new started, build up their knowledge of the design patterns starting from the most granular pieces building up into the larger pieces, then how to contribute.

**Getting Started**
- Introduction
- React & Storybook
- Themes
- Sandbox (demos, I kind of liked the play area context of sandbox, like anyone could hop in and play around with it. _Stolen from Cloudfour._).

**Design Elements**
- Spacing & Sizing
- Colors
- Tyopography
- Icons

**Utilities**
- all the utilities

**Components**
- all the components

**Help Docs**
- contributing guide

### CODE REVIEW

File: src/pages/react.hbs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+<p class="u-fontSizeLg u-spaceBottomMd u-padBottomMd u-borderBottom">
+  For ease of use we're actively creating a React component library that is directly 
+  based on the CSS components and utilities outlined here.
+</p>
+
+<img src="/getting-started/storybook.png" class="u-border">
+
+<p class='u-padTopNone'>
+  This will be uploaded to another Netlify site in the near future, but for now to 
+  access Storybook simplly close <a href="https://github.com/teamsnap/teamsnap-ui">TeamSnap UI</a>
```
Comment (saralohr) [inline src/pages/react.hbs]: SP "simply", "clone", "inside"

File: src/pages/index.hbs [hunk 2]
```diff
@@ -1,50 +1,228 @@
...
-</dl>
+<section class='doc-Intro'>
+  <h2 class="doc-Intro-header">Goals</h2>
+  <div class="Grid Grid--withGutter">
+    <div class="Grid-cell u-size1of3">
+      <div class="doc-IconBox">
+        <img src="/getting-started/consistency@1.5.png">
+      </div>
+      <h3>Consistency</h3>
+      <p>Across all of TeamSnap's web presence by providing clear guidelines and unifyed design patterns, along with reusable CSS and React toolkits.</p>
```
[the remark this answers is not in the document]
Comment (williamdingwall) [reply src/pages/index.hbs]: updated

File: src/pages/index.hbs [hunk 3]
```diff
@@ -1,50 +1,228 @@
...
+</section>
+
+<section class='doc-Intro'>
+  <h2 class="doc-Intro-header">Using TeamSnap UI</h2>
+  <h3>CDN</h3>
+  <p class="u-padTopSm u-padBottomSm">
+    For the fastest and simplest use, you can easily just drop in our CDN link to the &#60;head&#62; of your document.
+    Themes are also accessible by referencing the themes directory and theme file at the end of the URL <code>/css/themes/league.css</code>.
+  </p>
+  <pre class="language-markup u-padMd" data-bg="fill"><code>&#60;link rel="stylesheet" type="text/css" href="https://dugout.teamsnap.com/teamsnap-ui/2.15.1/css/teamsnap-ui.css"&#62;</code></pre>
```
[the remark this answers is not in the document]
Comment (williamdingwall) [reply src/pages/index.hbs]: I put a link in to the change log, that work for you?

File: src/pages/index.hbs [hunk 4]
```diff
@@ -1,50 +1,228 @@
...
+  color: #1a801a;
+  background-color: #fdfdfd;
+  width: 120px;
+}</code></pre>
+</section>
+
+<section class='doc-Intro'>
+  <h2 class="doc-Intro-header">Utility Classes</h2>
+  <p class="u-padTopSm u-padBottomSm">
+    Similar to the vairable utilities, TeamSnap UI also offers a wide range of Utility classes to easily make teaks
```
[the remark this answers is not in the document]
Comment (williamdingwall) [reply src/pages/index.hbs]: thanks, updated

File: src/pages/index.hbs [hunk 5]
```diff
@@ -1,50 +1,228 @@
...
+}
+  </code></pre>
+</section>
+
+<section class='doc-Intro'>
+  <h2 class="doc-Intro-header">Components</h2>
+  <p class="doc-Intro-block u-padTopSm u-padBottomSm">
+    As with any CSS Framework, TeamSnap UI comes with a slew of design patterns built out in HTML and CSS. 
+    TeamSnap UI's CSS components are primarily built of a top level class like "Button", then modifier classes
+    are added to create variations of that component like in the case below with "Button--primary". .Button
```
[the remark this answers is not in the document]
Comment (williamdingwall) [reply src/pages/index.hbs]: updated

File: src/pages/react.hbs (hunk 1 above)
Comment (williamdingwall) [reply src/pages/react.hbs]: updated

File: src/pages/index.hbs [hunk 6]
```diff
@@ -1,50 +1,228 @@
...
+  <small><b>Don't</b></small>
+  <pre class="language-css u-padMd" data-bg="fill"><code>
+.new-class { 
+  padding: 20px;
+  color: blue;
+}</code></pre>
+  <p class="u-padTopSm u-padBottomSm">
+    Basing styles on a set of SASS variables and functions allows for a lot of flexibility in updating design patterns across all 
+    TeamSnap's web presence or in specific sections, by overriding variables and compiling a new stylesheet. Stepping outside this
+    pattern for static values makes theming more difficult since those styles need to be managed in those soecific place or on specific
```
[the remark this answers is not in the document]
Comment (williamdingwall) [reply src/pages/index.hbs]: updated

File: src/pages/index.hbs (hunk 3 above)
[the remark this answers is not in the document]
Comment (saralohr) [reply src/pages/index.hbs]: Works for me!

File: src/pages/index.hbs [hunk 7]
```diff
@@ -1,50 +1,228 @@
...
+    Utilities are all tied back to the core set of styles and fall in line with the spacing, color, and typography 
+    scales so as they grow and change in the future or based on theme, so two will the tweaks made along the way. 
+    <a href="https://github.com/teamsnap/teamsnap-ui/tree/master/src/css/utils">Utility classes</a> should be viewed 
+    as a direct replacement for inline styling and extra classes to accomplish minor tweaks.
+  </p>
+  <small><b>Do</b>:</small>
+  <pre class="language-markup u-padMd" data-bg="fill"><code>&#60;p class="u-padTopMd u-spaceMd u-borderTop"&#62;Lorem ipsulm dalor sit amet&#60;/p&#62; </code></pre>
+  <small><b>Don't</b>:</small>
+  <pre class="language-markup u-padMd" data-bg="fill"><code>&#60;p style="padding-top: 16px; margin-top: 16px; border-top: 1px solid #d6d6d6;"&#62;Lorem ipsulm dalor sit amet&#60;/p&#62; </code></pre>
+  <small><b>Don't</b>:</small>
```
[the remark this answers is not in the document]
Comment (williamdingwall) [reply src/pages/index.hbs]: yeah that's a good call - updated... also added positive to the Do. A lot clearer. Thanks

**Comment by williamdingwall** [comment]
@saralohr Thanks!  and thanks for the review / help with some proof reading!

**Pull request merged**